### PR TITLE
Moving initialization of required services to PrioritizedOnAfterOpenProject

### PR DESCRIPTION
**Pull request by davkean**
This includes https://github.com/dotnet/project-system/pull/3443, so only review https://github.com/dotnet/project-system/pull/3447/commits/71016e11686b994d21238f3cb48576c369260813 onwards.

Fixes: https://github.com/dotnet/project-system/issues/3414.

### CODE REVIEW

**Comment by davkean** [comment]
Ugh, this isn't enough - seems like we're getting beaten to the punch by other code.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IUnconfiguredProjectTasksService.cs [hunk 1]
```diff
@@ -36,5 +69,32 @@ internal interface IUnconfiguredProjectTasksService
...
+        /// <typeparam name="T">
+        ///     The type of value returned by the joinable.
+        /// </typeparam>
+        /// <param name="action">
+        ///     The action to execute before the project is considered loaded in the host.
+        /// </param>
+        /// <exception cref="OperationCanceledException">
+        ///     Thrown if the project was already unloaded before this method was invoked.
+        /// </exception>
+        Task<T> PrioritizedLoadedInHostAsync<T>(Func<Task<T>> action);
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IUnconfiguredProjectTasksService.cs]: One of these is called `PrioritizedLoadedInHostAsync`, and the other `PrioritizedProjectLoadedInHostAsync`?  Aren't they meant to be overloads?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IUnconfiguredProjectTasksService.cs]: Yes they are supposed to be, fixing.

**Comment by Pilchie** [comment]
Tagging @jasonmalinowski who might be interested in this, and @lifengl for another review.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UnconfiguredProjectTasksService.cs [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
-        public UnconfiguredProjectTasksService([Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService)
+        public UnconfiguredProjectTasksService([Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService, IProjectThreadingService threadingService, ILoadedInHostListener loadedInHostListener)
         {
+            _prioritizedTaskQueue = new JoinableTaskQueue(threadingService.JoinableTaskContext);
             _tasksService = tasksService;
+            _threadingService = threadingService;
+            _loadedInHostListener = loadedInHostListener;
+        }
+
+        [ProjectAutoLoad(completeBy:ProjectLoadCheckpoint.ProjectFactoryCompleted)]
```
[the remark this answers is not in the document]
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UnconfiguredProjectTasksService.cs]: "LanguageService" and "CSharp" are not dynamic capabilities.

**Comment by Pilchie** [comment]
Approved to merge via [606841](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/606841)

**Pull request merged**